### PR TITLE
Fix mentions rendering in comment editor

### LINKED ISSUE #35047: [Bug]: comment_editor text field shows UID instead of user mention_bubble
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

The comment field doesn't resolve the @ user mention to the user mention bubble when still typing.

Only when commenting is the name resolved.

image

### Steps to reproduce

1. Open file sidebar
2. open comment tab
3. @ mention a user and keep typing
4. see UID instead of mention bubble

Seen on 24, 25 and 26 dev.

### Expected behavior

The username is replaced with the bubble when selected from a dropdown

### Installation method

Other Community project

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

_No response_

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
-
```

### List of activated Apps

```shell
-
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by artonge**
`NcRichContentEditable` needs an index of users to properly display them[0].
This commit adds a caching logic and provides it to `NcRichContentEditable`.

[0] https://github.com/nextcloud/nextcloud-vue/blob/01dc781d18e2d2a81fed6ad578b90532709b7e1d/src/mixins/richEditor/index.js#L106-L119

Fix https://github.com/nextcloud/server/issues/35047

Screenshot from 2022-11-16 14-18-05

### CODE REVIEW

**Comment by danxuliu** [comment]
This is already addressed in #35053, but it was not reviewed yet :-)

**Comment by artonge** [comment]
Closing in favor of https://github.com/nextcloud/server/pull/35053

**Comment by artonge** [comment]
Reopening after discussion with @danxuliu :)

**Review by miaulalala — APPROVED** [review]
LGTM

**Review by danxuliu — APPROVED** [review]
Tested and works :+1: 

One thing, though. It seems that editing a comment that ends with a mention in Firefox does not work (or, at least, it does not work here :-) ):
- Use Firefox
- Send a new comment which is just a mention (like `@user0`)
- Edit that comment
- The cursor does not appear, and typing is not reflected in the comment. On the other hand, if the message was `hello @user0` it is possible to place the cursor at the beginning with the mouse and type there, but not at the end.

But I guess this is not a problem caused by this pull request (well, it is because now there are rendered mentions, but you know what I mean ;-) ).

File: apps/comments/src/views/Comments.vue [hunk 1]
```diff
@@ -91,9 +92,9 @@ import RefreshIcon from 'vue-material-design-icons/Refresh'
 import MessageReplyTextIcon from 'vue-material-design-icons/MessageReplyText'
 import AlertCircleOutlineIcon from 'vue-material-design-icons/AlertCircleOutline'
 
-import Comment from '../components/Comment'
-import getComments, { DEFAULT_LIMIT } from '../services/GetComments'
-import cancelableRequest from '../utils/cancelableRequest'
+import Comment from '../components/Comment.vue'
+import getComments, { DEFAULT_LIMIT } from '../services/GetComments.js'
+import cancelableRequest from '../utils/cancelableRequest.js'
```
[the remark this answers is not in the document]
Comment (artonge) [reply apps/comments/src/views/Comments.vue:97]: Manually added as not having them leads to compilation warning.

This was introduced by: https://github.com/nextcloud/eslint-config/pull/292 :)

[the remark this answers is not in the document]
Comment (artonge) [reply apps/comments/src/views/Comments.vue:97]: File extension is not necessary for modules.
Also, Github might not display warnings for non-altered lines.

**Pull request merged**